src/socket_handle.c

```diff
@@ -200,36 +204,36 @@ struct initgroups_res {
 	gid_t *grps;
 };
 
-enum nss_status nss_getkey(uint32_t reqtype, void *fn, void *key, void *res, char *buf, size_t n, int *ret)
+static enum nss_status nss_getkey(uint32_t reqtype, struct mod_passwd *mod_passwd, struct mod_group *mod_group, void *key, void *res, char *buf, size_t n, int *ret)
 {
-	nss_getgrnam_r fn_grnam;
-	nss_getgrgid_r fn_grgid;
-	nss_getpwnam_r fn_pwnam;
-	nss_getpwuid_r fn_pwuid;
 	int retval = NSS_STATUS_UNAVAIL;
 	struct initgroups_res *initgroups_res;
+
+	if(ISPWREQ(reqtype)) assert(mod_passwd);
```

Comment:
This needs a comment above it explaining that this is for debugging only.

Reply:
Added a comment now.